client/jetpack-cloud/components/missing-payment-notification/index.tsx

```diff
@@ -14,18 +13,15 @@ export default function MissingPaymentNotification() {
 		return null;
 	}
 
-	const latestUnpaidInvoice = partner.keys.reduce< Invoice | null >( ( latestInvoice, key ) => {
-		if ( key.latestInvoice && key.latestInvoice.status === 'open' ) {
-			return key.latestInvoice;
-		}
-		return latestInvoice;
-	}, null );
+	const firstUnpaidInvoice = partner.keys.find(
+		( key ) => key.latestInvoice && key.latestInvoice.status === 'open'
+	)?.latestInvoice;
 
-	if ( latestUnpaidInvoice ) {
+	if ( firstUnpaidInvoice ) {
```

Comment:
Technically it's not necessarily the first (chronologically) unpaid invoice either, but instead an arbitrary unpaid invoice.

Reply:
firstUnpaidInvoice that my search function will find, but I agree, naming variables is an art. Thought about naming this 'unpaidInvoice', but this could lead people to think that there is just one unpaid invoice, when there can be multiple.